front/lib/plans/stripe.ts

```diff
@@ -290,3 +290,81 @@ export async function cancelSubscriptionImmediately({
 
   return true;
 }
+
+const REPORT_USAGE_VALUES = ["MAU_1", "MAU_5", "MAU_10", "PER_SEAT"];
+export type StripeSubscriptionValidityType =
+  | { valid: true }
+  | { valid: false; message?: string };
+
+/**
+ * Checks that a subscription created in Stripe is usable by Dust
+ */
+export function checkStripeSubscriptionValid(
+  stripeSubscription: Stripe.Subscription
+): StripeSubscriptionValidityType {
+  if (stripeSubscription.items.data.length === 0) {
+    return {
+      valid: false,
+      message: "Subscription has no items.",
+    };
+  }
+
+  // very unlikely, so handling is overkill at time of writing
+  if (stripeSubscription.items.has_more) {
+    return {
+      valid: false,
+      message: "Subscription has too many items.",
+    };
+  }
+
+  // All the business logic checks below are validating that the stripe
+  // subscription doesn't have a configuration that we don't support
+  for (const item of stripeSubscription.items.data) {
+    if (item.deleted) continue;
```

Comment:
```suggestion
    if (item.deleted) {
      continue;
    }
```

Reply:
Ahaha :upside_down_face: Let's have the team decide this one once and for all